airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -170,13 +174,22 @@ private CancellationScope generateSyncWorkflowRunnable(final ConnectionUpdaterIn
 
         if (workflowState.isFailed()) {
           reportFailure(connectionUpdaterInput, standardSyncOutput);
+
+          final FailureType failureType = standardSyncOutput.getFailures().isEmpty() ? null : standardSyncOutput.getFailures().get(0).getFailureType();
+          System.out.println(failureType);
+          if (failureType == FailureType.CONFIG_ERROR) {
+            // In the case that the failure is attributable to config_error, we will not retry again
+            // TODO: In the UI, the job still appears to be "retrying".  The Job is state=incomplete and the attempt=failed
+            workflowState.setFailed(true);
+            workflowState.setRetryFailedActivity(false);
+            cancellableSyncWorkflow.cancel();
+          } else {
+            prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
+          }
```

Comment:
The goal here is if the failureType == FailureType.CONFIG_ERROR (e.g. the user did something wrong) we don't need to retry this job with a second attempt.  How can we update the job from this point in the codebase?